src/services/firebase.ts

```diff
@@ -41,14 +46,43 @@ export module ourFirebase {
   export function setGamesList() {
     assertLoggedIn();
     // TODO: implement.
-    db().ref('TODO').once('value', gotGamesList);
+    db()
+      .ref('TODO')
+      .once('value', gotGamesList);
   }
 
   // Eventually dispatches the action setMatchesList
   // every time this field is updated:
   //  /gamePortal/gamePortalUsers/$myUserId/privateButAddable/matchMemberships
   export function listenToMyMatchesList() {
     // TODO: implement
+    assertLoggedIn();
+    db()
+      .ref(
+        'gamePortal/gamePortalUsers' +
+          testUser +
+          '/privateButAddable/matchMemberships'
+      )
+      .on('value', getMatchLists);
+  }
+
+  function getMatchLists(snap: firebase.database.DataSnapshot) {
+    let updateMatchList: any = new Array();
+    snap.forEach(function(childSnapshot: firebase.database.DataSnapshot) {
+      let matchId = childSnapshot.key;
+      updateMatchList.push(getMatchDetail(matchId));
+      return false;
+    });
+    store.dispatch(updateMatchList);
+  }
+
+  function getMatchDetail(matchId: any) {
```

Comment:
Add types, this will help you find mistakes, e.g., you return a promise (not MatchInfo) from here.